internal/atlassian/data_jira_issues.go

```diff
@@ -0,0 +1,170 @@
+					}),
+				},
+				{
+					Name:       "jql",
+					Type:       cty.String,
+					Doc:        "A JQL expression. For performance reasons, this field requires a bounded query. A bounded query is a query with a search restriction.",
+					ExampleVal: cty.StringVal("order by key desc"),
+				},
+				{
+					Name:         "properties",
+					Type:         cty.List(cty.String),
+					Doc:          "A list of up to 5 issue properties to include in the results.",
+					MaxInclusive: cty.NumberIntVal(5),
+				},
+				{
+					Name:         "size",
+					Type:         cty.Number,
+					Doc:          "Size limit to retrieve.",
+					MinInclusive: cty.NumberIntVal(0),
+					Constraints:  constraint.NonNull,
+					DefaultVal:   cty.NumberIntVal(0),
+				},
+			},
+		},
+	}
+}
+
+func searchJiraIssuesData(loader ClientLoadFn) plugin.RetrieveDataFunc {
+	return func(ctx context.Context, params *plugin.RetrieveDataParams) (plugindata.Data, diagnostics.Diag) {
+		cli, err := parseConfig(params.Config, loader)
+		if err != nil {
+			return nil, diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to parse configuration",
+			}}
+		}
+		req, err := parseSearchIssuesReq(params.Args)
+		if err != nil {
+			return nil, diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to parse arguments",
+			}}
+		}
+		size := 0
+		if attr := params.Args.GetAttrVal("size"); !attr.IsNull() {
+			num, _ := attr.AsBigFloat().Int64()
+			size = int(num)
+		}
+		var issues plugindata.List
+		for {
+			res, err := cli.SearchIssues(ctx, req)
+			if err != nil {
+				return nil, handleClientError(err)
+			}
+			for _, v := range res.Issues {
+				data, err := plugindata.ParseAny(v)
+				if err != nil {
+					return nil, diagnostics.Diag{{
+						Severity: hcl.DiagError,
+						Summary:  "Failed to parse data",
+					}}
+				}
+				issues = append(issues, data)
+				if size > 0 && len(issues) == size {
+					break
+				}
+			}
+			if (size > 0 && len(issues) == size) || res.NextPageToken == nil {
+				break
+			}
+			req.NextPageToken = res.NextPageToken
+		}
+		return issues, nil
+	}
+}
+
+func parseSearchIssuesReq(args *dataspec.Block) (*client.SearchIssuesReq, error) {
+	if args == nil {
+		return nil, fmt.Errorf("arguments are required")
+	}
+	req := &client.SearchIssuesReq{}
+	if attr := args.GetAttrVal("expand"); !attr.IsNull() {
+		req.Expand = client.String(attr.AsString())
+	}
+	if attr := args.GetAttrVal("jql"); !attr.IsNull() {
+		req.JQL = client.String(attr.AsString())
+	}
+	if attr := args.GetAttrVal("fields"); !attr.IsNull() {
+		for _, field := range attr.AsValueSlice() {
+			if field.IsNull() {
+				continue
+			}
+			req.Fields = append(req.Fields, field.AsString())
+		}
+	}
+	if attr := args.GetAttrVal("properties"); !attr.IsNull() {
+		for _, property := range attr.AsValueSlice() {
+			if property.IsNull() {
+				continue
+			}
+			req.Properties = append(req.Properties, property.AsString())
+		}
+	}
```

Comment:
... then this could be simplified

```suggestion
	for _, property := range args.GetAttrVal("properties").AsValueSlice() {
		if property.IsNull() {
			continue
		}
		req.Properties = append(req.Properties, property.AsString())
	}
```